cmd/admin-handlers.go

```diff
@@ -1768,3 +1772,128 @@ func (a adminAPIHandlers) KMSKeyStatusHandler(w http.ResponseWriter, r *http.Req
 	}
 	writeSuccessResponseJSON(w, resp)
 }
+
+// ServiceInfoHandler - GET /minio/admin/v1/serviceinfo?type={serviceType}
+// ----------
+// Get service information
+func (a adminAPIHandlers) ServiceInfoHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := newContext(r, w, "ServiceInfo")
+
+	objectAPI := validateAdminReq(ctx, w, r)
+	if objectAPI == nil {
+		return
+	}
+	var jsonBytes []byte
+	var err error
+
+	vars := mux.Vars(r)
+
+	switch madmin.InfoType(vars[madmin.INFO]) {
+	case madmin.ETCD:
+		errorString := ""
+		if globalEtcdClient == nil {
+			writeErrorResponseJSON(ctx, w, errorCodes.ToAPIErr(ErrETCDNotConfigured), r.URL)
+			return
+		}
+		etcdEndPoints := globalEtcdClient.Endpoints()
+		var etcdInfos []madmin.ETCDInfo
+		for _, endPoint := range etcdEndPoints {
+
+			if err := checkConnection(endPoint); err != nil {
+				errorString = errEtcdUnreachable.Error()
+			}
+
+			etcdInfo := madmin.ETCDInfo{
+				ETCDEndpoint: endPoint,
+				Domain:       Environment.Get(EnvMinioDomain, ""),
+				PublicIPs:    globalDomainIPs,
+				Error:        errorString,
+			}
+			etcdInfos = append(etcdInfos, etcdInfo)
+		}
+		// Marshal API response
+		jsonBytes, err = json.Marshal(etcdInfos)
+		if err != nil {
+			writeErrorResponseJSON(ctx, w, toAdminAPIErr(ctx, err), r.URL)
+			return
+		}
+
+	case madmin.LOGGER:
+		var loggerInfo []madmin.ServerLoggerInfo
+		if endpoint, ok := Environment.Lookup(EnvLoggerHTTPEndpoint); ok {
+			result := madmin.ServerLoggerInfo{Endpoint: endpoint, Enabled: true}
+			if e := checkConnection(endpoint); e != nil {
+				result.Error = e.Error()
+			}
+			loggerInfo = append(loggerInfo, result)
+
+		} else {
+			for _, http := range globalServerConfig.Logger.HTTP {
+				result := madmin.ServerLoggerInfo{Endpoint: http.Endpoint, Enabled: http.Enabled}
+				if http.Enabled {
+					e := checkConnection(http.Endpoint)
+					if e != nil {
+						result.Error = e.Error()
+					}
+				}
+				loggerInfo = append(loggerInfo, result)
+			}
+		}
+
+		// Marshal API response
+		jsonBytes, err = json.Marshal(loggerInfo)
+		if err != nil {
+			writeErrorResponseJSON(ctx, w, toAdminAPIErr(ctx, err), r.URL)
+			return
+		}
+
+	case madmin.AUDIT:
+		var auditInfo madmin.ServerAuditInfo
+		if endpoint, ok := Environment.Lookup(EnvAuditLoggerHTTPEndpoint); ok {
+			auditInfo = madmin.ServerAuditInfo{Endpoint: endpoint, Enabled: true}
+			if e := checkConnection(endpoint); e != nil {
+				auditInfo.Error = e.Error()
+			}
+		}
+
+		// Marshal API response
+		jsonBytes, err = json.Marshal(auditInfo)
+		if err != nil {
+			writeErrorResponseJSON(ctx, w, toAdminAPIErr(ctx, err), r.URL)
+			return
+		}
+	case madmin.LAMBDA:
+		config, err := readServerConfig(ctx, objectAPI)
```

Comment:
There will be no need to read the server config if we enhance the TargetInfo as suggested. We will get these details from MarshalJSON()

Reply:
removed.